protos/flyteidl/datacatalog/datacatalog.proto

```diff
@@ -150,6 +157,22 @@ message ListDatasetsResponse {
     string next_token = 2;
 }
 
+message GetOrReserveArtifactRequest {
```

Comment:
also can you rebase? I add a bunch of documentation to all the structures, you can get that too :)

Reply:
Yeah i have merged master